src/usr/local/www/diag_packet_capture.php

```diff
@@ -479,7 +482,14 @@ function fixup_host($value, $position) {
 	}
 
 	if ($port != "") {
-		$matches[] = "port ".fixup_not($port);
+		for($i = 0; $i < count($ports); $i++){
+                    	 if($i == count($ports)-1) {
+                	                $ports_matches .= "port ".($ports[$i]);
+               	         } else {
+                	                $ports_matches .= "port ".($ports[$i]). " or ";
```

Comment:
Same as above, appears to be missing `fixup_not`, and needs spaces around the `.`(s).

Reply:
now same boolean logic as for hosts - 
and, or, not

debug output sample:
/usr/sbin/tcpdump -i vtnet1 -p -c 100 -s 0 -w /root/packetcapture.cap 'ip and tcp and ( port 22 and port 80 and port not 100 or port not 50) and ( host 1.1.1.1 and host 8.8.8.8)'